### ISSUE TITLE
Unable to build shared libraries

### ISSUE THREAD

**Issue by jmcarcell** [xianyi/OpenBLAS#3979]
If I build with shared enabled, I get errors that look like this:
```
ar: `u' modifier ignored since `D' is the default (see `U')
ranlib ../../../libopenblas_skylakexp-r0.3.22.dev.a
bfd plugin: ../../../libopenblas_skylakexp-r0.3.22.dev.a: lseek: Invalid argument
bfd plugin: ../../../libopenblas_skylakexp-r0.3.22.dev.a: lseek: Invalid argument
bfd plugin: ../../libopenblas_skylakexp-r0.3.22.dev.a: lseek: Invalid argument
bfd plugin: ../../libopenblas_skylakexp-r0.3.22.dev.a: lseek: Invalid argument
make[3]: Leaving directory '/home/jmcarcell/Test/OpenBLAS/lapack-netlib/LAPACKE/utils'
make[2]: Leaving directory '/home/jmcarcell/Test/OpenBLAS/lapack-netlib/LAPACKE'
make[1]: Leaving directory '/home/jmcarcell/Test/OpenBLAS/lapack-netlib'
ar: ../../libopenblas_skylakexp-r0.3.22.dev.a: error reading lapacke_ztr_trans.o: file truncated
make[1]: *** [../../Makefile.tail:46: libs] Error 1
make[1]: Leaving directory '/home/jmcarcell/Test/OpenBLAS/driver/level2'
make: *** [Makefile:184: libs] Error 1
```

I reduced the make command to this: `make -j48 shared` for the previous example. With this make command 
```
'make' '-j16' 'MAKE_NB_JOBS=0' 'ARCH=x86_64' 'DYNAMIC_ARCH=1' 'DYNAMIC_OLDER=1' 'TARGET=GENERIC' 'USE_LOCKING=1' 'USE_OPENMP=0' 'USE_THREAD=0' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'
```
then there are errors like this one:
```
8093:make[2]: *** [../../Makefile.tail:46: libs] Bus error (core dumped)
```

System: Arch Linux with gcc 12.2.1, CPU: Intel Xeon Silver 4214. Any ideas on what is going wrong? More generally, I'm finding this problem when building using spack, which has `shared` enabled by default. Full build log:
[build.log](https://github.com/xianyi/OpenBLAS/files/11103296/build.log)

Checking the build log I realize there are more errors:
```
ar  -ru ../../libopenblas_skylakexp-r0.3.22.dev.a sgbmv_n.o sgbmv_t.o ssbmv_U.o ssbmv_L.o sspmv_U.o sspmv_L.o sspr_U.o sspr_L.o sspr2_U.o sspr2_L.o ssyr_U.o ssyr_L.o ssyr2_U.o ssyr2_L.o stbmv_NUU.o stbmv_NUN.o stbmv_NLU.o stbmv_NLN.o stbmv_TUU.o stbmv_TUN.o stbmv_TLU.o stbmv_TLN.o stbsv_NUU.o stbsv_NUN.o stbsv_NLU.o stbsv_NLN.o stbsv_TUU.o stbsv_TUN.o stbsv_TLU.o stbsv_TLN.o stpmv_NUU.o stpmv_NUN.o stpmv_NLU.o stpmv_NLN.o stpmv_TUU.o stpmv_TUN.o stpmv_TLU.o stpmv_TLN.o stpsv_NUU.o stpsv_NUN.o stpsv_NLU.o stpsv_NLN.o stpsv_TUU.o stpsv_TUN.o stpsv_TLU.o stpsv_TLN.o strmv_NUU.o strmv_NUN.o strmv_NLU.o strmv_NLN.o strmv_TUU.o strmv_TUN.o strmv_TLU.o strmv_TLN.o strsv_NUU.o strsv_NUN.o strsv_NLU.o strsv_NLN.o strsv_TUU.o strsv_TUN.o strsv_TLU.o strsv_TLN.o sgemv_thread_n.o sgemv_thread_t.o sger_thread.o ssymv_thread_U.o ssymv_thread_L.o ssyr_thread_U.o ssyr_thread_L.o ssyr2_thread_U.o ssyr2_thread_L.o sspr_thread_U.o sspr_thread_L.o sspr2_thread_U.o sspr2_thread_L.o strmv_thread_NUU.o strmv_thread_NUN.o strmv_thread_NLU.o strmv_thread_NLN.o strmv_thread_TUU.o strmv_thread_TUN.o strmv_thread_TLU.o strmv_thread_TLN.o sspmv_thread_U.o sspmv_thread_L.o stpmv_thread_NUU.o stpmv_thread_NUN.o stpmv_thread_NLU.o stpmv_thread_NLN.o stpmv_thread_TUU.o stpmv_thread_TUN.o stpmv_thread_TLU.o stpmv_thread_TLN.o sgbmv_thread_n.o sgbmv_thread_t.o ssbmv_thread_U.o ssbmv_thread_L.o stbmv_thread_NUU.o stbmv_thread_NUN.o stbmv_thread_NLU.o stbmv_thread_NLN.o stbmv_thread_TUU.o stbmv_thread_TUN.o stbmv_thread_TLU.o stbmv_thread_TLN.o dgbmv_n.o dgbmv_t.o dsbmv_U.o dsbmv_L.o dspmv_U.o dspmv_L.o dspr_U.o dspr_L.o dspr2_U.o dspr2_L.o dsyr_U.o dsyr_L.o dsyr2_U.o dsyr2_L.o dtbmv_NUU.o dtbmv_NUN.o dtbmv_NLU.o dtbmv_NLN.o dtbmv_TUU.o dtbmv_TUN.o dtbmv_TLU.o dtbmv_TLN.o dtbsv_NUU.o dtbsv_NUN.o dtbsv_NLU.o dtbsv_NLN.o dtbsv_TUU.o dtbsv_TUN.o dtbsv_TLU.o dtbsv_TLN.o dtpmv_NUU.o dtpmv_NUN.o dtpmv_NLU.o dtpmv_NLN.o dtpmv_TUU.o dtpmv_TUN.o dtpmv_TLU.o dtpmv_TLN.o dtpsv_NUU.o dtpsv_NUN.o dtpsv_NLU.o dtpsv_NLN.o dtpsv_TUU.o dtpsv_TUN.o dtpsv_TLU.o dtpsv_TLN.o dtrmv_NUU.o dtrmv_NUN.o dtrmv_NLU.o dtrmv_NLN.o dtrmv_TUU.o dtrmv_TUN.o dtrmv_TLU.o dtrmv_TLN.o dtrsv_NUU.o dtrsv_NUN.o dtrsv_NLU.o dtrsv_NLN.o dtrsv_TUU.o dtrsv_TUN.o dtrsv_TLU.o dtrsv_TLN.o dgemv_thread_n.o dgemv_thread_t.o dger_thread.o dsymv_thread_U.o dsymv_thread_L.o dsyr_thread_U.o dsyr_thread_L.o dsyr2_thread_U.o dsyr2_thread_L.o dspr_thread_U.o dspr_thread_L.o dspr2_thread_U.o dspr2_thread_L.o dtrmv_thread_NUU.o dtrmv_thread_NUN.o dtrmv_thread_NLU.o dtrmv_thread_NLN.o dtrmv_thread_TUU.o dtrmv_thread_TUN.o dtrmv_thread_TLU.o dtrmv_thread_TLN.o dspmv_thread_U.o dspmv_thread_L.o dtpmv_thread_NUU.o dtpmv_thread_NUN.o dtpmv_thread_NLU.o dtpmv_thread_NLN.o dtpmv_thread_TUU.o dtpmv_thread_TUN.o dtpmv_thread_TLU.o dtpmv_thread_TLN.o dgbmv_thread_n.o dgbmv_thread_t.o dsbmv_thread_U.o dsbmv_thread_L.o dtbmv_thread_NUU.o dtbmv_thread_NUN.o dtbmv_thread_NLU.o dtbmv_thread_NLN.o dtbmv_thread_TUU.o dtbmv_thread_TUN.o dtbmv_thread_TLU.o dtbmv_thread_TLN.o cgbmv_n.o cgbmv_t.o cgbmv_r.o cgbmv_c.o cgbmv_o.o cgbmv_u.o cgbmv_s.o cgbmv_d.o chbmv_U.o chbmv_L.o chbmv_V.o chbmv_M.o cher_U.o cher_L.o cher_V.o cher_M.o cher2_U.o cher2_L.o cher2_V.o cher2_M.o chpmv_U.o chpmv_L.o chpmv_V.o chpmv_M.o chpr_U.o chpr_L.o chpr_V.o chpr_M.o chpr2_U.o chpr2_L.o chpr2_V.o chpr2_M.o csbmv_U.o csbmv_L.o cspr2_U.o cspr2_L.o csyr2_U.o csyr2_L.o ctbmv_NUU.o ctbmv_NUN.o ctbmv_NLU.o ctbmv_NLN.o ctbmv_TUU.o ctbmv_TUN.o ctbmv_TLU.o ctbmv_TLN.o ctbmv_RUU.o ctbmv_RUN.o ctbmv_RLU.o ctbmv_RLN.o ctbmv_CUU.o ctbmv_CUN.o ctbmv_CLU.o ctbmv_CLN.o ctbsv_NUU.o ctbsv_NUN.o ctbsv_NLU.o ctbsv_NLN.o ctbsv_TUU.o ctbsv_TUN.o ctbsv_TLU.o ctbsv_TLN.o ctbsv_RUU.o ctbsv_RUN.o ctbsv_RLU.o ctbsv_RLN.o ctbsv_CUU.o ctbsv_CUN.o ctbsv_CLU.o ctbsv_CLN.o ctpmv_NUU.o ctpmv_NUN.o ctpmv_NLU.o ctpmv_NLN.o ctpmv_TUU.o ctpmv_TUN.o ctpmv_TLU.o ctpmv_TLN.o ctpmv_RUU.o ctpmv_RUN.o ctpmv_RLU.o ctpmv_RLN.o ctpmv_CUU.o ctpmv_CUN.o ctpmv_CLU.o ctpmv_CLN.o ctpsv_NUU.o ctpsv_NUN.o ctpsv_NLU.o ctpsv_NLN.o ctpsv_TUU.o ctpsv_TUN.o ctpsv_TLU.o ctpsv_TLN.o ctpsv_RUU.o ctpsv_RUN.o ctpsv_RLU.o ctpsv_RLN.o ctpsv_CUU.o ctpsv_CUN.o ctpsv_CLU.o ctpsv_CLN.o ctrmv_NUU.o ctrmv_NUN.o ctrmv_NLU.o ctrmv_NLN.o ctrmv_TUU.o ctrmv_TUN.o ctrmv_TLU.o ctrmv_TLN.o ctrmv_RUU.o ctrmv_RUN.o ctrmv_RLU.o ctrmv_RLN.o ctrmv_CUU.o ctrmv_CUN.o ctrmv_CLU.o ctrmv_CLN.o ctrsv_NUU.o ctrsv_NUN.o ctrsv_NLU.o ctrsv_NLN.o ctrsv_TUU.o ctrsv_TUN.o ctrsv_TLU.o ctrsv_TLN.o ctrsv_RUU.o ctrsv_RUN.o ctrsv_RLU.o ctrsv_RLN.o ctrsv_CUU.o ctrsv_CUN.o ctrsv_CLU.o ctrsv_CLN.o cspmv_U.o cspmv_L.o cspr_U.o cspr_L.o csyr_U.o csyr_L.o cgemv_thread_n.o cgemv_thread_t.o cgemv_thread_r.o cgemv_thread_c.o cgemv_thread_o.o cgemv_thread_u.o cgemv_thread_s.o cgemv_thread_d.o cger_thread_U.o cger_thread_C.o cger_thread_V.o cger_thread_D.o csymv_thread_U.o csymv_thread_L.o chemv_thread_U.o chemv_thread_L.o chemv_thread_V.o chemv_thread_M.o csyr_thread_U.o csyr_thread_L.o cher_thread_U.o cher_thread_L.o cher_thread_V.o cher_thread_M.o csyr2_thread_U.o csyr2_thread_L.o cher2_thread_U.o cher2_thread_L.o cher2_thread_V.o cher2_thread_M.o cspr_thread_U.o cspr_thread_L.o chpr_thread_U.o chpr_thread_L.o chpr_thread_V.o chpr_thread_M.o cspr2_thread_U.o cspr2_thread_L.o chpr2_thread_U.o chpr2_thread_L.o chpr2_thread_V.o chpr2_thread_M.o ctrmv_thread_NUU.o ctrmv_thread_NUN.o ctrmv_thread_NLU.o ctrmv_thread_NLN.o ctrmv_thread_TUU.o ctrmv_thread_TUN.o ctrmv_thread_TLU.o ctrmv_thread_TLN.o ctrmv_thread_RUU.o ctrmv_thread_RUN.o ctrmv_thread_RLU.o ctrmv_thread_RLN.o ctrmv_thread_CUU.o ctrmv_thread_CUN.o ctrmv_thread_CLU.o ctrmv_thread_CLN.o cspmv_thread_U.o cspmv_thread_L.o chpmv_thread_U.o chpmv_thread_L.o chpmv_thread_V.o chpmv_thread_M.o ctpmv_thread_NUU.o ctpmv_thread_NUN.o ctpmv_thread_NLU.o ctpmv_thread_NLN.o ctpmv_thread_TUU.o ctpmv_thread_TUN.o ctpmv_thread_TLU.o ctpmv_thread_TLN.o ctpmv_thread_RUU.o ctpmv_thread_RUN.o ctpmv_thread_RLU.o ctpmv_thread_RLN.o ctpmv_thread_CUU.o ctpmv_thread_CUN.o ctpmv_thread_CLU.o ctpmv_thread_CLN.o cgbmv_thread_n.o cgbmv_thread_t.o cgbmv_thread_r.o cgbmv_thread_c.o cgbmv_thread_o.o cgbmv_thread_u.o cgbmv_thread_s.o cgbmv_thread_d.o csbmv_thread_U.o csbmv_thread_L.o chbmv_thread_U.o chbmv_thread_L.o chbmv_thread_V.o chbmv_thread_M.o ctbmv_thread_NUU.o ctbmv_thread_NUN.o ctbmv_thread_NLU.o ctbmv_thread_NLN.o ctbmv_thread_TUU.o ctbmv_thread_TUN.o ctbmv_thread_TLU.o ctbmv_thread_TLN.o ctbmv_thread_RUU.o ctbmv_thread_RUN.o ctbmv_thread_RLU.o ctbmv_thread_RLN.o ctbmv_thread_CUU.o ctbmv_thread_CUN.o ctbmv_thread_CLU.o ctbmv_thread_CLN.o zgbmv_n.o zgbmv_t.o zgbmv_r.o zgbmv_c.o zgbmv_o.o zgbmv_u.o zgbmv_s.o zgbmv_d.o zhbmv_U.o zhbmv_L.o zhbmv_V.o zhbmv_M.o zher_U.o zher_L.o zher_V.o zher_M.o zher2_U.o zher2_L.o zher2_V.o zher2_M.o zhpmv_U.o zhpmv_L.o zhpmv_V.o zhpmv_M.o zhpr_U.o zhpr_L.o zhpr_V.o zhpr_M.o zhpr2_U.o zhpr2_L.o zhpr2_V.o zhpr2_M.o zsbmv_U.o zsbmv_L.o zspmv_U.o zspmv_L.o zspr_U.o zspr_L.o zspr2_U.o zspr2_L.o zsyr_U.o zsyr_L.o zsyr2_U.o zsyr2_L.o ztbmv_NUU.o ztbmv_NUN.o ztbmv_NLU.o ztbmv_NLN.o ztbmv_TUU.o ztbmv_TUN.o ztbmv_TLU.o ztbmv_TLN.o ztbmv_RUU.o ztbmv_RUN.o ztbmv_RLU.o ztbmv_RLN.o ztbmv_CUU.o ztbmv_CUN.o ztbmv_CLU.o ztbmv_CLN.o ztbsv_NUU.o ztbsv_NUN.o ztbsv_NLU.o ztbsv_NLN.o ztbsv_TUU.o ztbsv_TUN.o ztbsv_TLU.o ztbsv_TLN.o ztbsv_RUU.o ztbsv_RUN.o ztbsv_RLU.o ztbsv_RLN.o ztbsv_CUU.o ztbsv_CUN.o ztbsv_CLU.o ztbsv_CLN.o ztpmv_NUU.o ztpmv_NUN.o ztpmv_NLU.o ztpmv_NLN.o ztpmv_TUU.o ztpmv_TUN.o ztpmv_TLU.o ztpmv_TLN.o ztpmv_RUU.o ztpmv_RUN.o ztpmv_RLU.o ztpmv_RLN.o ztpmv_CUU.o ztpmv_CUN.o ztpmv_CLU.o ztpmv_CLN.o ztpsv_NUU.o ztpsv_NUN.o ztpsv_NLU.o ztpsv_NLN.o ztpsv_TUU.o ztpsv_TUN.o ztpsv_TLU.o ztpsv_TLN.o ztpsv_RUU.o ztpsv_RUN.o ztpsv_RLU.o ztpsv_RLN.o ztpsv_CUU.o ztpsv_CUN.o ztpsv_CLU.o ztpsv_CLN.o ztrmv_NUU.o ztrmv_NUN.o ztrmv_NLU.o ztrmv_NLN.o ztrmv_TUU.o ztrmv_TUN.o ztrmv_TLU.o ztrmv_TLN.o ztrmv_RUU.o ztrmv_RUN.o ztrmv_RLU.o ztrmv_RLN.o ztrmv_CUU.o ztrmv_CUN.o ztrmv_CLU.o ztrmv_CLN.o ztrsv_NUU.o ztrsv_NUN.o ztrsv_NLU.o ztrsv_NLN.o ztrsv_TUU.o ztrsv_TUN.o ztrsv_TLU.o ztrsv_TLN.o ztrsv_RUU.o ztrsv_RUN.o ztrsv_RLU.o ztrsv_RLN.o ztrsv_CUU.o ztrsv_CUN.o ztrsv_CLU.o ztrsv_CLN.o zgemv_thread_n.o zgemv_thread_t.o zgemv_thread_r.o zgemv_thread_c.o zgemv_thread_o.o zgemv_thread_u.o zgemv_thread_s.o zgemv_thread_d.o zger_thread_U.o zger_thread_C.o zger_thread_V.o zger_thread_D.o zsymv_thread_U.o zsymv_thread_L.o zhemv_thread_U.o zhemv_thread_L.o zhemv_thread_V.o zhemv_thread_M.o zsyr_thread_U.o zsyr_thread_L.o zher_thread_U.o zher_thread_L.o zher_thread_V.o zher_thread_M.o zsyr2_thread_U.o zsyr2_thread_L.o zher2_thread_U.o zher2_thread_L.o zher2_thread_V.o zher2_thread_M.o zspr_thread_U.o zspr_thread_L.o zhpr_thread_U.o zhpr_thread_L.o zhpr_thread_V.o zhpr_thread_M.o zspr2_thread_U.o zspr2_thread_L.o zhpr2_thread_U.o zhpr2_thread_L.o zhpr2_thread_V.o zhpr2_thread_M.o ztrmv_thread_NUU.o ztrmv_thread_NUN.o ztrmv_thread_NLU.o ztrmv_thread_NLN.o ztrmv_thread_TUU.o ztrmv_thread_TUN.o ztrmv_thread_TLU.o ztrmv_thread_TLN.o ztrmv_thread_RUU.o ztrmv_thread_RUN.o ztrmv_thread_RLU.o ztrmv_thread_RLN.o ztrmv_thread_CUU.o ztrmv_thread_CUN.o ztrmv_thread_CLU.o ztrmv_thread_CLN.o zspmv_thread_U.o zspmv_thread_L.o zhpmv_thread_U.o zhpmv_thread_L.o zhpmv_thread_V.o zhpmv_thread_M.o ztpmv_thread_NUU.o ztpmv_thread_NUN.o ztpmv_thread_NLU.o ztpmv_thread_NLN.o ztpmv_thread_TUU.o ztpmv_thread_TUN.o ztpmv_thread_TLU.o ztpmv_thread_TLN.o ztpmv_thread_RUU.o ztpmv_thread_RUN.o ztpmv_thread_RLU.o ztpmv_thread_RLN.o ztpmv_thread_CUU.o ztpmv_thread_CUN.o ztpmv_thread_CLU.o ztpmv_thread_CLN.o zgbmv_thread_n.o zgbmv_thread_t.o zgbmv_thread_r.o zgbmv_thread_c.o zgbmv_thread_o.o zgbmv_thread_u.o zgbmv_thread_s.o zgbmv_thread_d.o zsbmv_thread_U.o zsbmv_thread_L.o zhbmv_thread_U.o zhbmv_thread_L.o zhbmv_thread_V.o zhbmv_thread_M.o ztbmv_thread_NUU.o ztbmv_thread_NUN.o ztbmv_thread_NLU.o ztbmv_thread_NLN.o ztbmv_thread_TUU.o ztbmv_thread_TUN.o ztbmv_thread_TLU.o ztbmv_thread_TLN.o ztbmv_thread_RUU.o ztbmv_thread_RUN.o ztbmv_thread_RLU.o ztbmv_thread_RLN.o ztbmv_thread_CUU.o ztbmv_thread_CUN.o ztbmv_thread_CLU.o ztbmv_thread_CLN.o
ar: `u' modifier ignored since `D' is the default (see `U')
ranlib ../../libopenblas_skylakexp-r0.3.22.dev.a
ar: ../../libopenblas_skylakexp-r0.3.22.dev.a: error reading zgetrs_C_parallel.o: file truncated
make[1]: *** [../../Makefile.tail:46: libs] Error 1
```

This warning seems to appear always: 
```
ar: `u' modifier ignored since `D' is the default (see `U')
```

**Comment by martin-frbg** [comment]
Are you compiling on a local filesystem, or something mounted ? (worst case: host-shared folder in a vm)

**Comment by jmcarcell** [comment]
Local filesystem and native

**Comment by martin-frbg** [comment]
Are you using version 4.4 of make by any chance ? There have been some reports of it requiring `--jobserver-style=pipe` with many existing makefiles

**Comment by thrasibule** [comment]
could you try #3981 ? I don't have a machine with enough cores to trigger this, but according to here: https://www.gnu.org/software/make/manual/html_node/Archive-Pitfalls.html maybe this helps. It should also remove the warnings at the very least.

**Comment by jmcarcell** [comment]
@martin-frbg Yes I'm using make 4.4. I tried passing that option but it doesn't seem to change anything
@thrasibule I tried but `make -j 48 shared` gives this error, which it's very possible I have found it before:
```
ar: ../../libopenblas_skylakexp-r0.3.22.dev.a: error reading dsyevr_2stage.o: file truncated
make[2]: *** [../../Makefile.tail:46: libs] Error 1
make[2]: Leaving directory '/home/jmcarcell/Test/OpenBLAS/lapack/trtri'
make[1]: *** [Makefile:10: libs] Error 1
make[1]: Leaving directory '/home/jmcarcell/Test/OpenBLAS/lapack'
make: *** [Makefile:184: libs] Error 1
make: *** Waiting for unfinished jobs....
ranlib ../../../libopenblas_skylakexp-r0.3.22.dev.a
```
Do you think this is because of the parallel build? I will let it build now with a single core and report the results later or tomorrow. Certainly the warnings disappear. 
EDIT: Full build log below (different build attempt, this one core dumped with #3981...):
[build-branch.log](https://github.com/xianyi/OpenBLAS/files/11113542/build-branch.log)

**Comment by jmcarcell** [comment]
`make -j 1 shared` seems to be working fine (develop). I also checked htop and it's using all the cores at several steps in the build like compiling fortran (not the only one), see screenshot below


![2023-03-30-180400_3715x753_scrot](https://user-images.githubusercontent.com/22276694/228898193-fdbeb0b7-bde7-435f-b549-099977f13e84.png)

EDIT: And `make -j 2 shared` doesn't work with similar errors as before

**Comment by thrasibule** [comment]
this is interesting: 
```
make[1]: warning: -j48 forced in submake: resetting jobserver mode
```

Basically there are two levels of parallelism being applied: the -j48 that you pass on the command line, and the make command is redefined to be make -j48 in Makefile.conf. How about if you try ```MAKE_NB_JOBS=0 make -j48 shared```, does it build fine?

**Comment by jmcarcell** [comment]
Same errors as before with `MAKE_NB_JOBS=0 make -j48 shared`

**Comment by martin-frbg** [comment]
We´ve seen this before - #3899 : what happens when you either leave off the `libs netlib shared` (for a full build including the handful of tests), or run `MAKE_NB_JOBS=48 make shared` ?

**Comment by jmcarcell** [comment]
`MAKE_NB_JOBS=48 make shared` works
Building `libs netlib shared` was what was being built when I found the error (this is the default in spack) but I reduced it to `shared` since building `libs` or `netlib` only was fine.  For the tests I've run, building with `libs netlib shared` has the same result as building with `shared` only

**Comment by martin-frbg** [comment]
Thanks. I was just worried that the `libs netlib shared` stanza - why is everybody so keen on avoiding the tests - might be less protected by makefile rules than the default `all`. Need to check if it is some inadvertent recent change that caused the difference in `make -j` behaviour, or a different implementation of the option from gmake 4.4 onwards.

**Comment by thrasibule** [comment]
So I think we need to add shared to the .NOTPARALLEL prerequisite. that way when we do make shared, that will force lib and netlib to be run sequentially.
Separately there is some unclear interaction between MAKE_NB_JOBS and running make with a -j parameter or  (having MAKEFLAGS=jx) in the environment. The way the main makefile is written, we want it to be run sequentially and then the sub makes with be called with a -j flag according to MAKE_NB_JOBS setting. I think it'd be more consistent to have MAKE_NB_JOBS overwrite MAKEFLAGS (that would also prevent these warnings (```make[1]: warning: -j48 forced in submake: resetting jobserver mode```)
I think #3983 should be an improvement. I've verified that libs and netlib run serially now, but I can't trigger the write corruptions with my machine.

**Comment by jmcarcell** [comment]
`make -j48 shared` doesn't work with #3983 (core dumped, should it work?), see log below. Let me know if you need any more 'runs' that trigger the corruptions

[build.log](https://github.com/xianyi/OpenBLAS/files/11122545/build.log)

**Comment by thrasibule** [comment]
Are you sure #3983 was applied? And you did a make clean before rerunning? Your log still shows ```make[1]: warning: -j48 forced in submake: resetting jobserver mode``` which I don't think should happen if the patch was applied. I was hoping that would fix it, but I guess I don't understand anything then...

**Comment by jmcarcell** [comment]
Sorry you are right (when cleaning I reseted back to develop by mistake). It does work fine with #3983

**Comment by martin-frbg** [comment]
closing as fixed by #3983
